### DISCUSSION TITLE
WSLg uses Remote Desktop Connections?

### DISCUSSION THREAD

**Discussion by davidshen84** [microsoft/wslg#687] [Q&A] [answered]
Hi,

I found this strange correlated behaviour between WSLg applications and the `mstsc.exe` process.

- When I start up my Windows 11, there's a `mstsc.exe` process started automatically and I cannot kill it.
- If I start a WSLg application, like Emacs GUI, the *Remote Desktop Connection* disappeared from the *Processes* tab in the task manager, but there's still a `mstsc.exe` process in the *Details* tab.
- If I kill the `mstsc.exe` process, my WSLg application got force restart as well.

Another strange observation is about Docker Desktop. If I start Docker Desktop, there are 3 `mstsc.exe` processes started automatically and cannot be killed.

All these ghost `mstsc` processes disappear after I disable WSLg on my system.

Is it true that WSLg uses the `mstsc` processes?

**Comment by Biswa96** [comment, marked as answer]
> Is it true that WSLg uses the mstsc processes?

Yes. More details can be found in source code, see https://github.com/microsoft/wslg/blob/main/WSLGd/main.cpp. `/mnt/c/Windows/System32/mstsc.exe` is executed with LaunchProcess function.

> If I kill the mstsc.exe process, my WSLg application got force restart as well.

Do not kill the mstsc process abruptly. It may cause issue with running WSL instance. In the https://github.com/microsoft/wslg/blob/main/WSLGd/ProcessMonitor.cpp file, LaunchProcess is called in a loop. Hence, that mstsc process is restarted automatically.

**Comment by hello-smile6** [comment]
Would mstsc work for remote access?

**Reply by Biswa96** [reply]
Theoretically it is possible. There are some discussion about that here https://github.com/microsoft/wslg/issues/470

**Comment by idg10** [comment]
The huge problem with the 3 that Docker starts is that they frequently get into a state where they steal the desktop focus every couple of minutes.

**Comment by idg10** [comment]
Ah... but I have since discovered that installed 2H22 of Windows 11 doesn't appear to bring all parts of WSL2 up to date. After going to https://aka.ms/wslstorepage and installing the latest bits from there, all the focus stealing issues seem to have stopped.

**Reply by GF-Huang** [reply]
Yes, I updated it. But it lag my keyboard typing, and start `msrdc.exe` repeatlly.


https://user-images.githubusercontent.com/4510984/209905028-272f71dd-9df9-4728-9579-2b253e6be518.mp4
